src/RateChangeQueue.sol

```diff
@@ -39,6 +39,13 @@ library RateChangeQueue {
         require(queue.head < queue.tail, "Queue is empty");
         return queue.changes[queue.head];
     }
+    
+    function peekTail(
+        Queue storage queue
+    ) internal view returns (RateChange memory) {
```

Comment:
@Stebalien Added this to fix the bug with repeated rate changes in a single epoch.